internal/services/instance.go

```diff
@@ -44,6 +45,17 @@ func (s *Instance) CreateInstance(ctx context.Context, ownerID uint, projectName
 		return "", fmt.Errorf("failed to get project: %w", err)
 	}
 
+	// validate the instances array is not empty
+	if len(instances) == 0 {
+		return "", fmt.Errorf("at least one instance is required")
+	}
+
+	// validate the providers
+	for _, instance := range instances {
```

Comment:
This should be added to the `Validate` method called in the instance handler.

Reply:
@MSevey I did a bit of digging and founf out there is an existing provider validation. I think it'd be best to reuse that, so i have updated the PR